modules/apps/portal-workflow/portal-workflow-kaleo-test/src/testIntegration/java/com/liferay/portal/workflow/kaleo/internal/runtime/integration/test/WorkflowTaskManagerImplTest.java

```diff
@@ -798,6 +803,54 @@ public void testAssignApproveWorkflowBlogsEntryAsPortalContentReviewer()
 		_deactivateWorkflow(BlogsEntry.class.getName(), 0, 0);
 	}
 
+	@Test
+	public void testBrokenScriptedAssignment() throws Exception {
+		String workflowDefinitionName =
+			"Broken Scripted Assignment Workflow Definition";
+
+		_createScriptedAssignmentWorkflow(
+			"broken-scripted-assignment-workflow-definition.json",
+			workflowDefinitionName);
+
+		_activateWorkflow(
+			0, BlogsEntry.class.getName(), 0, 0, workflowDefinitionName, 1);
+
+		BlogsEntry blogsEntry = null;
+
+		try (LogCapture logCapture = LoggerTestUtil.configureLog4JLogger(
+				"com.liferay.portal.workflow.kaleo.runtime.internal.petra." +
+					"executor.GraphWalkerPortalExecutor",
+				LoggerTestUtil.ERROR)) {
+
+			blogsEntry = _addBlogsEntry();
+
+			List<LogEntry> logEntries = logCapture.getLogEntries();
+
+			Assert.assertEquals(logEntries.toString(), 1, logEntries.size());
+		}
+
+		Assert.assertEquals(
+			WorkflowConstants.STATUS_PENDING, blogsEntry.getStatus());
+
+		WorkflowInstance workflowInstance = _getWorkflowInstance(
+			BlogsEntry.class.getName(), blogsEntry.getEntryId());
+
+		Assert.assertTrue(workflowInstance.isComplete());
+
+		Assert.assertEquals(
+			1,
+			_kaleoLogLocalService.getKaleoInstanceKaleoLogsCount(
+				_company.getCompanyId(),
+				workflowInstance.getWorkflowInstanceId(),
+				new ArrayList<Integer>() {
+					{
+						add(WorkflowLog.WORKFLOW_INSTANCE_FAIL);
+					}
+				}));
+
+		_deactivateWorkflow(0, BlogsEntry.class.getName(), 0, 0);
+	}
```

Comment:
@pedro-oliveira446 move this test to `WorkflowInstanceManagerImplTest` and rename to `testCompleteKaleoInstanceOnFailure`.

Thanks!

Reply:
Done!